src/crypto/tls_mbedtls_alt.c

```diff
@@ -121,7 +121,9 @@ typedef mbedtls_tls_prf_types int;
  * then checking the client cert against the DN hint in certificate verify_cb.
  * Comment out to disable feature and remove ~3k from binary .text segment */
 #define TLS_MBEDTLS_CERT_VERIFY_EXTMATCH
-//#define TLS_MBEDTLS_CERT_DISABLE_KEY_USAGE_CHECK
+#ifdef MBEDTLS_ALLOW_PRIVATE_ACCESS
+#define TLS_MBEDTLS_CERT_DISABLE_KEY_USAGE_CHECK
```

Comment:
Can you please share a reference to this define? I couldn't find it anywhere, is this NXP internal? And also how is it reliant on `MBEDTLS_ALLOW_PRIVATE_ACCESS`? As per docs, the private access using this is deprecated and if you still need access then this has to be raised upstream.

Reply:
TLS_MBEDTLS_CERT_DISABLE_KEY_USAGE_CHECK is used inside this file to disable key usage check, which is private access.

Reply:
Ah sorry missed it, I did a search and couldn't find it but was on the wrong branch, thanks.

Reply:
Can you please also elaborate on how does the flags `MBEDTLS_X509_EXT_KEY_USAGE` and `MBEDTLS_X509_EXT_EXTENDED_KEY_USAGE` tie in to private access, I see that `key_usage` is private in  `mbedtls_x509_crt` but don't  see that used in this file? and how does the reporting to AP work?

Reply:
Thanks for the clarification, one last question, why the dependency on `MBEDTLS_ALLOW_PRIVATE_ACCESS`? the `key_usage` is marked as private in `mbedtls_x509_crt` but unused by the hostap.

Reply:
tls_mbedtls_verify_cb for TLS_MBEDTLS_CERT_DISABLE_KEY_USAGE_CHECK,
it disables MBEDTLS_X509_EXT_KEY_USAGE and MBEDTLS_X509_EXT_EXTENDED_KEY_USAGE of ext_types,
to disable key usage check, which is MBEDTLS_PRIVATE.
So for applications not defining MBEDTLS_ALLOW_PRIVATE_ACCESS, we don't enable TLS_MBEDTLS_CERT_DISABLE_KEY_USAGE_CHECK to avoid build error.

Reply:
```
#if defined(TLS_MBEDTLS_CERT_DISABLE_KEY_USAGE_CHECK)
    crt->ext_types &= ~MBEDTLS_X509_EXT_KEY_USAGE;
    crt->ext_types &= ~MBEDTLS_X509_EXT_EXTENDED_KEY_USAGE;
#endif
```
I only see above changes and both are enumerations, isn't private access limited to structure members?

Reply:
Hi @krish2718 any comments for this PR?